### PR TITLE
PostgreSQL support

### PR DESCRIPTION
**Pull request by zeuner**
Since the minetest master branch defines the table schema to use with the PostgreSQL backend, it makes sense to support it in auxiliary tools.

### CODE REVIEW

File: TileGenerator.cpp [hunk 1]
```diff
@@ -283,6 +286,16 @@ void TileGenerator::openDb(const std::string &input)
 
 	if(backend == "sqlite3")
 		m_db = new DBSQLite3(input);
+#if USE_POSTGRESQL
+	else if(backend == "postgresql") {
+		std::ifstream ifs((input + "/world.mt").c_str());
+		if(!ifs.good())
+			throw std::runtime_error("Failed to read world.mt");
+		std::string const connect_string = get_setting("pgsql_connection", ifs);
+		ifs.close();
```
Comment (sfan5) [inline TileGenerator.cpp]: These lines should be in `db-postgresql.cpp`

File: db-postgresql.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+
+DBPostgreSQL::~DBPostgreSQL()
+{
+	try {
+		checkResults(PQexec(db, "COMMIT;"));
+	} catch (
+		std::exception& caught
+	) {
```
Comment (sfan5) [inline db-postgresql.cpp]: `} catch(std::exception &e) {` one single line please

File: db-postgresql.cpp [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+		std::cerr << "could not finalize: " << caught.what();
+		std::cerr << std::endl;
```
Comment (sfan5) [inline db-postgresql.cpp]: this can also be moved to the above line

File: db-postgresql.cpp [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+	PGresult *results = execPrepared(
+		"get_block_pos", 0,
+		NULL, NULL, NULL, false, false
+	);
+
+	int numrows = PQntuples(results);
+
+	for (int row = 0; row < numrows; ++row) {
+		positions.push_back(pg_to_blockpos(results, row, 0));
+	}
```
Comment (sfan5) [inline db-postgresql.cpp]: brace removal possible

**Comment by zeuner** [comment]
@sfan5: Thanks. I like these.

**Pull request merged**